database/rrdset.c

```diff
@@ -1904,6 +1904,8 @@ void rrdset_done(RRDSET *st) {
                             metalog_commit_delete_dimension(rd);
                         } else {
                             /* Do not delete this dimension */
+                            last = rd;
```

Comment:
Why are we checking if dimensions can be deleted at all in the case of the dbengine? The deletion would occur at the dbrotation, this whole scan of the dimensions seems to be redundant now in the case of the dbengine.

Reply:
It used to be that `OBSOLETE` dimensions were deleted. Now for `dbengine` this changes to a new state, that is `OBSOLETE` will be converted to `ARCHIVED` and the collector state is torn down. Rotation does indeed delete them in a different code path.

Reply:
Yes, so do we need to scan the dimensions here to look for possible deletions when using dbengine?

Reply:
It's not deletions, it's state changes from `OBSOLETE` to `ARCHIVED`. The condition is the same but instead of deleting we archive them. Otherwise the dimensions would remain forever at the `OBSOLETE` state which would be wrong.

Reply:
Ok, this makes sense. Very general question - why do we need `#ifdef ENABLE_DBENGINE` around blocks such as this?

Reply:
`dbengine` still isn't a required component for netdata so we need to support binaries that are built without it.

Reply:
Yes, but why do we need to protect this block with a `#ifdef` ? The `if` could be enough.

Reply:
Some pieces of code breaks compilation such as `metalog_commit_delete_dimension()`.